portfolio/src/main/java/com/google/sps/servlets/TranslationServlet.java

```diff
@@ -0,0 +1,39 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.cloud.translate.Translate;
+import com.google.cloud.translate.TranslateOptions;
+import com.google.cloud.translate.Translation;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/translate")
+public class TranslationServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    String[] comments;
+    String languageCode = request.getParameter("languageCode");
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
+
+    comments = request.getParameterValues("comments");
+
+    // Translate and store each comment.
```

Comment:
Does it make sense to extract translation code into a separate helper function? This way doPost will be focused on extracting parameters from request and serializing response, while the helper function will be focused on obtaining Translate service and, well, translation.

Reply:
Yup, this sounds like a great idea to me. Should this function translate a single piece of text at a time? To me this seems like a more modular and reusable approach, but that would mean every piece of text has to get the translate service instance every time. My other thought would be to accept a list of strings and translate them all to return a list of strings, which would solve having to re-initialize the service for every piece of text but if I had only one comment I'd have to convert it to an array for no reason.

Reply:
The decision depends on the project, in some project translating list of strings is better, in another translating individual string makes more sense. 

In this particular case my thoughts were: 
- I'd like this code to be readable 3 months from now
- I want it to not waste too many resources in a typical case

Breaking out into separate function helps with readability. When you read doPost you know that details of translation are hidden and you can abstract from them, focusing on understanding interactions with servlet. When you read translate implementation, http related logic is nowhere around and doesn't draw your attention.

For this project typical case seems to be list of comments. It makes sense to design API and optimize for this assumption. It's Ok to create single element array for rare case. That's why I think here translate function should take list of comments and language code, create service instance once and use it to translate all strings.

That being said... This is a toy project. We are just using it to play around with though process that would happen in the real project. Personally for me it is more important to raise this thought than to make you write code in a particular way. Feel free to pick any option or just leave the code as is. I'm fine either way.